### PR TITLE
Update build dependencies for 3.0.0

### PR DESCRIPTION
**Pull request by kgaillot**
... along with some other build changes

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens , could you review this when you get a chance? Also, I had problems running a lab on this due to logging issues on my test cluster, but I didn't see any problems related to this PR. Could you also run a lab on this just to be sure it's not causing trouble somehow? Thanks ...

File: daemons/controld/controld_control.c [hunk 1]
```diff
@@ -127,7 +127,7 @@ void
 crmd_fast_exit(crm_exit_t exit_code)
 {
     if (pcmk_is_set(controld_globals.fsa_input_register, R_STAYDOWN)) {
-        crm_warn("Inhibiting respawn "CRM_XS" remapping exit code %d to %d",
+        crm_warn("Inhibiting respawn "QB_XS" remapping exit code %d to %d",
```
Comment (clumens) [inline daemons/controld/controld_control.c:130]: Everywhere else, we put spaces between the double quotes and CRM_XS.

File: daemons/controld/controld_election.c [hunk 2]
```diff
@@ -185,7 +185,7 @@ feature_update_callback(xmlNode * msg, int call_id, int rc, xmlNode * output, vo
     if (rc != pcmk_ok) {
         fsa_data_t *msg_data = NULL;
 
-        crm_notice("Feature update failed: %s "CRM_XS" rc=%d",
+        crm_notice("Feature update failed: %s "QB_XS" rc=%d",
```
Comment (clumens) [inline daemons/controld/controld_election.c:188]: Same here.

File: daemons/controld/controld_fsa.c [hunk 3]
```diff
@@ -505,7 +505,7 @@ s_crmd_fsa_actions(fsa_data_t * fsa_data)
 
             /* Error checking and reporting */
         } else {
-            crm_err("Action %s not supported "CRM_XS" %#llx",
+            crm_err("Action %s not supported "QB_XS" %#llx",
```
Comment (clumens) [inline daemons/controld/controld_fsa.c:508]: And here.

File: daemons/controld/controld_join_client.c [hunk 4]
```diff
@@ -278,7 +278,7 @@ do_cl_join_finalize_respond(long long action,
 
     if (was_nack) {
         crm_err("Shutting down because cluster join with leader %s failed "
-                CRM_XS" join-%d NACK'd", welcome_from, join_id);
+                QB_XS" join-%d NACK'd", welcome_from, join_id);
```
Comment (clumens) [inline daemons/controld/controld_join_client.c:281]: Both places in this file.

File: daemons/controld/controld_te_actions.c [hunk 5]
```diff
@@ -402,7 +402,7 @@ execute_rsc_action(pcmk__graph_t *graph, pcmk__graph_action_t *action)
         no_wait = TRUE;
     }
 
-    crm_notice("Initiating %s operation %s%s on %s%s "CRM_XS" action %d",
+    crm_notice("Initiating %s operation %s%s on %s%s "QB_XS" action %d",
```
Comment (clumens) [inline daemons/controld/controld_te_actions.c:405]: And here.

File: daemons/pacemakerd/pacemakerd.c [hunk 6]
```diff
@@ -412,7 +412,7 @@ main(int argc, char **argv)
         }
     }
 
-    crm_notice("Starting Pacemaker %s "CRM_XS" build=%s features:%s",
+    crm_notice("Starting Pacemaker %s "QB_XS" build=%s features:%s",
```
Comment (clumens) [inline daemons/pacemakerd/pacemakerd.c:415]: And here.

File: daemons/pacemakerd/pcmkd_subdaemons.c [hunk 7]
```diff
@@ -890,7 +890,7 @@ stop_child(pcmk_child_t * child, int signal)
 
     errno = 0;
     if (kill(child->pid, signal) == 0) {
-        crm_notice("Stopping %s "CRM_XS" sent signal %d to process %lld",
+        crm_notice("Stopping %s "QB_XS" sent signal %d to process %lld",
```
Comment (clumens) [inline daemons/pacemakerd/pcmkd_subdaemons.c:893]: And here.

File: lib/common/mainloop.c [hunk 8]
```diff
@@ -255,7 +255,7 @@ crm_signal_dispatch(GSource *source, GSourceFunc callback, gpointer userdata)
     crm_signal_t *sig = (crm_signal_t *) source;
 
     if(sig->signal != SIGCHLD) {
-        crm_notice("Caught '%s' signal "CRM_XS" %d (%s handler)",
+        crm_notice("Caught '%s' signal "QB_XS" %d (%s handler)",
```
Comment (clumens) [inline lib/common/mainloop.c:258]: And here.

File: m4/PKG_CHECK_VAR.m4
Comment (clumens) [inline m4/PKG_CHECK_VAR.m4:1]: This should also be removed from EXTRA_DIST in `Makefile.am`.

**Comment by clumens** [comment]
> @clumens , could you review this when you get a chance? Also, I had problems running a lab on this due to logging issues on my test cluster, but I didn't see any problems related to this PR. Could you also run a lab on this just to be sure it's not causing trouble somehow? Thanks ...

No regression test or CTSlab test failures, though I don't know why my build did not run into the same problem that CI did.  I definitely built and installed the correct thing.

File: daemons/controld/controld_control.c (hunk 1 above)
Comment (kgaillot) [reply daemons/controld/controld_control.c:130]: good point

File: m4/PKG_CHECK_VAR.m4
Comment (kgaillot) [reply m4/PKG_CHECK_VAR.m4:1]: thanks

**Comment by kgaillot** [comment]
updated CRM_XS and pkg-config commits per review

**Comment by kgaillot** [comment]
rebased on current main to resolve conflicts

File: rpm/pacemaker.spec.in [hunk 9]
```diff
@@ -55,11 +55,11 @@
 # Define conditionals so that "rpmbuild --with <feature>" and
 # "rpmbuild --without <feature>" can enable and disable specific features
 
-## Add option to enable support for stonith/external fencing agents
+## Add option for Linux-HA (stonith/external) fencing agent support
 %if 0%{?suse_version} > 0
-%bcond_without stonithd
+%bcond_without linuxha
```
Comment (nrwahl2) [inline rpm/pacemaker.spec.in:60]: Some part of this commit seems to have broken OpenSUSE builds:
```
15:03:16  make WITH="--with linuxha" rpm
15:03:16  make: *** No rule to make target 'linuxha"'.  Stop.
```

though it looks like a syntax issue (notice that the second double quote is included in the bogus make target name), so I'm not sure how `with stonithd` ever worked either

Comment (kgaillot) [reply rpm/pacemaker.spec.in:60]: Actually it's a companion PR in ci-tools that enabled `--with linuxha` for suse builds (we haven't been building that code in CI so far). That had two problems, ci-tools is dropping the quotes somewhere along the line so make thinks `linuxha"` is a make target, and I forgot to add the default `--without doc` to the suse string and it's failing on a doc dependency.

**Pull request merged**